.github/workflows/performance-tests.yml

```
@@ -0,0 +1,584 @@
+        uses: patrickedqvist/wait-for-vercel-preview@v1.3.1
+        id: wait-for-deployment
+        continue-on-error: true
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          max_timeout: 600  # Increased from 300s (5min) to 600s (10min) for larger deployments
+          check_interval: 15  # Increased from 10s to 15s to reduce API pressure
+          
+      - name: Fallback deployment detection
+        id: fallback-deployment
+        if: steps.wait-for-deployment.outcome == 'failure'
+        run: |
```

actionlint (1.7.7)

"github.head_ref" is potentially untrusted. avoid using it directly in inline scripts. instead, pass it through an environment variable. see https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions for more details

(expression)